irohad/validation/impl/chain_validator_impl.cpp

```diff
@@ -30,41 +20,64 @@ namespace iroha {
         : supermajority_checker_(supermajority_checker),
           log_(logger::log("ChainValidator")) {}
 
-    bool ChainValidatorImpl::validateBlock(
-        std::shared_ptr<shared_model::interface::Block> block,
-        ametsuchi::MutableStorage &storage) const {
-      log_->info("validate block: height {}, hash {}",
-                 block->height(),
-                 block->hash().hex());
-      auto check_block =
-          [this](const auto &block, auto &queries, const auto &top_hash) {
-            auto peers = queries.getLedgerPeers();
-            if (not peers) {
-              return false;
-            }
-            return block.prevHash() == top_hash
-                and supermajority_checker_->hasSupermajority(block.signatures(),
-                                                             peers.value());
-          };
-
-      // check inside of temporary storage
-      return storage.check(*block, check_block);
-    }
-
     bool ChainValidatorImpl::validateChain(
```

Comment:
This name is misleading: it also applies chain in successful case

Reply:
Created a task, since it is out of scope of this pull request